larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -681,12 +755,12 @@ void LArMCParticleHelper::SelectCaloHits(const CaloHitList *const pCaloHitList,
         {
             const MCParticle *const pHitParticle(MCParticleHelper::GetMainMCParticle(pCaloHit));
 
-            LArMCParticleHelper::MCRelationMap::const_iterator mcIter = mcToPrimaryMCMap.find(pHitParticle);
+            LArMCParticleHelper::MCRelationMap::const_iterator mcIter = mcToTargetMCMap.find(pHitParticle);
 
-            if (mcToPrimaryMCMap.end() == mcIter)
+            if (mcToTargetMCMap.end() == mcIter)
                 continue;
 
-            const MCParticle *const pPrimaryParticle = mcIter->second;
+            const MCParticle *const pPrimaryParticle = LArMCParticleHelper::GetPrimaryMCParticle(pHitParticle);
```

Comment:
Can you explain the reasoning for this change, thanks? If subtle, could add an // ATTN comment.

Reply:
In the folding case, for which the code was originally written, the function would be passed the map particle->primary particle. Then the code checks if the particle chain (from primary to particle) passes some checks - if there is a photon that travels beyond a certain distance or a neutron in the chain it does not pass the checks. When we are unfolding we can have the case where this function is instead passed the map particle->particle and so pPrimaryParticle may not be equal to mcIter->second. Therefore, I changed this line to get the PrimaryParticle from a function instead of the map. For half the cases it is a repetition as this has already been done when the user choses to fold back, so maybe I could add a statement that checks whether mcIter->first == mcIter->second and only run this function when this is true? and use mcIter->second when this is not true?